src/plugins/integrations/AWSIntegration.ts

```diff
@@ -529,23 +546,39 @@ class AWSIntegration implements Integration {
                             },
                         });
                     }
-
-                    const me = this;
                     const originalFunction = integration.wrappedFuncs[wrappedFunctionName];
-                    const wrappedCallback = (err: any, data: any) => {
-                        if (err && activeSpan) {
-                            activeSpan.setErrorTag(err);
-                        }
-                        if (data) {
-                            AWSIntegration.injectTraceLink(activeSpan, request, config);
-                        }
-                        if (activeSpan) {
-                            activeSpan.closeWithCallback(me, originalCallback, [err, data]);
-                        }
-                    };
 
-                    return originalFunction.apply(this, [wrappedCallback]);
+                    if (originalCallback) {
+                        const me = this;
+                        const wrappedCallback = (err: any, data: any) => {
+                            if (err && activeSpan) {
+                                activeSpan.setErrorTag(err);
+                            }
+                            if (data) {
+                                AWSIntegration.injectTraceLink(activeSpan, request, config);
+                            }
+                            if (activeSpan) {
+                                activeSpan.closeWithCallback(me, originalCallback, [err, data]);
+                            }
+                        };
 
+                        return originalFunction.apply(this, [wrappedCallback]);
+                    } else {
+                        request.on('error', (error: any) => {
+                            if (error && activeSpan) {
```

Comment:
Don't we close span on error?

Reply:
👍 Good catch. We should close the span in case of http error. Fixed.